### DISCUSSION TITLE
User_Setup file for XIAO ESP32-C3 with 320x240 display with ILI9341 controller

### DISCUSSION THREAD

**Discussion by Rollmops67** [Bodmer/TFT_eSPI#2957] [Compatible displays and setup files]
And yet another setup file :
[User_Setup_XIAO_ESP32-C3_ILI9341_TFT.zip](https://github.com/Bodmer/TFT_eSPI/files/13270076/User_Setup_XIAO_ESP32-C3_ILI9341_TFT.zip)

The XIAO ESP32-C3 has the advantage of being the size of a thumbnail, so it can be handy if you want to build a little display with the XIAO glued (with some double-sided adhesive) at the rear of the display.

The drawback is that is has only 11 GPIOs, and the display needs 6 of them.
So there are just 5 GPIOs remaining (3 of t.hem being also analog inputs) or 3 GPIOs + 2 GPIOs for a I2C bus.

Roland

**Comment by WigandClone** [comment]
**My User_Setup.h. CS and BLK are not connected.**

#define ILI9341_DRIVER

#define TFT_WIDTH  240
#define TFT_HEIGHT 320

#define TFT_MOSI  D10  // D10 (GPIO10)
#define TFT_SCLK   D8  // D8  (GPIO8)
#define TFT_DC     D2  // D0  (GPIO2)
#define TFT_RST    D3  // D1  (GPIO3)
// #define TFT_BL     D7  // D7  (GPIO20)
// #define TFT_CS    D4  // No CS

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_GFXFF
#define SMOOTH_FONT

#define SPI_FREQUENCY  10000000
#define USER_SETUP_LOADED


**But only working with an old core-lib**

; PlatformIO Project Configuration File

; Not working
; [env:seeed_xiao_esp32c3]
; platform = espressif32
; board = seeed_xiao_esp32c3
; framework = arduino
; lib_deps = bodmer/TFT_eSPI@^2.5.43
; build_flags =  -Ilib/TFT_eSPI

; Working
[env:seeed_xiao_esp32c3]
platform = espressif32@6.5.0
board = seeed_xiao_esp32c3
framework = arduino
board_build.arduino.core = esp32
board_build.arduino.version = 2.0.14

**Reply by Basitadam** [reply]
You can run it esp32-3.x.x with this file
[TFT_eSPI_ESP32_C3.zip](https://github.com/user-attachments/files/21507154/TFT_eSPI_ESP32_C3.zip)
